osmnames/import_osm/set_names.sql

```diff
@@ -3,35 +3,43 @@ CREATE FUNCTION get_alternative_names(all_tags HSTORE, name TEXT)
 RETURNS TEXT AS $$
 DECLARE
   alternative_names TEXT[];
+  alternative_names_string TEXT;
 BEGIN
   SELECT array_agg(DISTINCT(all_tags -> key))
   FROM unnest(akeys(all_tags)) AS key
-  WHERE (key LIKE 'name:%' OR key LIKE '%[_]name') AND (all_tags->key NOT ILIKE name)
+  WHERE (key LIKE 'name:%' OR key LIKE '%[_]name')
   INTO alternative_names;
-  RETURN COALESCE(array_to_string(alternative_names, ','), '');
+  alternative_names := array_remove(alternative_names, name);
+  alternative_names_string := array_to_string(alternative_names, ',');
+  RETURN COALESCE(regexp_replace(alternative_names_string, E'\\s+', ' ', 'g'), '');
 END;
 $$ LANGUAGE plpgsql IMMUTABLE;
 
 
 DROP FUNCTION IF EXISTS get_name(HSTORE);
 CREATE FUNCTION get_name(all_tags HSTORE)
 RETURNS TEXT AS $$
-  SELECT COALESCE(all_tags -> 'name:en',
+  SELECT COALESCE(
+                  all_tags -> 'name:en',
                   all_tags -> 'name:fr',
                   all_tags -> 'name:de',
                   all_tags -> 'name:es',
                   all_tags -> 'name:ru',
                   all_tags -> 'name:zh',
                   split_part(get_alternative_names(all_tags, ''), ',', 1),
-                  '');
+                  ''
+                  );
 $$ LANGUAGE 'sql' IMMUTABLE;
 
 
 UPDATE osm_linestring SET name = get_name(all_tags) WHERE name = '' IS NOT FALSE;
 UPDATE osm_polygon SET name = get_name(all_tags) WHERE name = '' IS NOT FALSE;
 UPDATE osm_point SET name = get_name(all_tags) WHERE name = '' IS NOT FALSE;
 
-
 UPDATE osm_linestring SET alternative_names = get_alternative_names(all_tags, name);
 UPDATE osm_polygon SET alternative_names = get_alternative_names(all_tags, name);
 UPDATE osm_point SET alternative_names = get_alternative_names(all_tags, name);
+
+UPDATE osm_linestring SET name =  regexp_replace(name, E'\\s+', ' ', 'g') WHERE name LIKE '%'||chr(9)||'%';
```

Comment:
Hi @philippks 
Thanks for the review! Yes, I thought about it as well. But we can't do that, since `get_name()` is only called for the rows where the `name` is null or an empty string.
```*.sql'
 UPDATE osm_linestring SET name = get_name(all_tags) WHERE name = '' IS NOT FALSE;
```
But we would like to remove tabs from names in all rows. Calling the `get_name()` function for all entries instead slows this down. Do you suggest that I rename the function to something like `get_name_if_empty_or_null()` to avoid confusion?